src/Site/Controller/Redirect.php

```diff
@@ -0,0 +1,85 @@
+<?php
```

Comment:
yikes man...the complexity is off the chart here...maybe there are parts here that actually belong in the UI or need to be resolved outside of this controller...not sure, I can just see there is way too much complexity in here.

Reply:
I've added some comments to explain the various redirect decisions, and reduced the complexity by getting rid of the AppModel instance. There are still a lot of if statements because there are multiple cases that need to be handled here:

* User logs in, but their most recent project was deleted by manager
* User logs in, but they have been removed from their most recent project
* User has an old session cookie with a most recent project ID in it, but their login has expired

And so on. But hopefully with the comments in my most recent commit the logic is comprehensible now.